lib/iris/util.py

```diff
@@ -2222,27 +2225,90 @@ def _strip_metadata_from_dims(cube, dims):
+        which offers richer shape handling.
+    """
+    message = (
+        "iris.util.mask_cube_from_shapefile has been deprecated, and will be removed in a "
+        "future release. Please use iris.util.mask_cube_from_shape instead."
+    )
+    warn_deprecated(message)
+    # Call the new function `mask_cube_from_shape`with the same parameters.
+    return mask_cube_from_shape(
+        cube,
+        shape,
+        shape_crs,
+        in_place=in_place,
+        minimum_weight=minimum_weight,
+        **kwargs,
+    )
+
+
+def mask_cube_from_shape(
+    cube: iris.cube.Cube,
+    shape: shapely.Geometry,
+    shape_crs: cartopy.crs | pyproj.CRS,
+    in_place: bool = False,
+    minimum_weight: float = 0.0,
+    **kwargs,
+) -> iris.cube.Cube | None:
+    """Mask all points in a cube that do not intersect a shape object.
+
+    Mask a :class:`~iris.cube.Cube` with any shape object, (e.g. Natural Earth Shapefiles via cartopy).
+    Finds the overlap between the `shape` and the :class:`~iris.cube.Cube` and
+    masks out any cells that __do not__ intersect the shape.
 
-    Finds the overlap between the `shape` and the `cube` in 2D xy space and
-    masks out any cells with less % overlap with shape than set.
-    Default behaviour is to count any overlap between shape and cell as valid
+    Shapes can be polygons, lines or points.
+
+    By default, all cells touched by geometries are kept (equivalent to `minimum_weight=0`). This behaviour
+    can be changed by increasing the `minimum_weight` keyword argument or setting `all_touched=False`,
+    then only the only cells whose center is within the polygon or that are selected by Bresenham’s line algorithm
+    (for line type shapes) are kept.   For points, the `minimum_weight` is ignored, and the cell that intersects the point
+    is kept.
 
     Parameters
     ----------
     cube : :class:`~iris.cube.Cube` object
         The `Cube` object to masked. Must be singular, rather than a `CubeList`.
-    shape : Shapely.Geometry object
+    shape : shapely.Geometry object
         A single `shape` of the area to remain unmasked on the `cube`.
         If it a line object of some kind then minimum_weight will be ignored,
         because you cannot compare the area of a 1D line and 2D Cell.
-    minimum_weight : float , default=0.0
-        A number between 0-1 describing what % of a cube cell area must
-        the shape overlap to include it.
+    shape_crs : cartopy.crs.CRS, default=None
+        The coordinate reference system of the shape object.
     in_place : bool, default=False
         Whether to mask the `cube` in-place or return a newly masked `cube`.
         Defaults to False.
+    minimum_weight : float, default=0.0
+        A number between 0-1 describing what % of a cube cell area must the shape overlap to be masked.
+        Only applied to polygon shapes.  If the shape is a line or point then this is ignored.
+
+    Other Parameters
```

Comment:
How did you decide which ones were `kwargs` and which ones were named parameters?

Reply:
I decided that `kwargs` would be parameters that are specifically used by `rasterio` functions only, but I'm happy to expose them if that's clearer?

Reply:
I'm not sure that's true of `invert`. And you are doing a little handling of `all_touched` too. It's not like the 10s of Matplotlib arguments which iris.plot passes straight down; I think in this case it would be clearer to have everything as normal `Parameters`.

Today I learned that Sphinx sticks `Parameters` and `Other Parameters` together when rendering 🤷